homeassistant/components/energyzero/sensor.py

```diff
@@ -0,0 +1,215 @@
+        value_fn=lambda data: data[SERVICE_ENERGY_TODAY].price_at_time(
+            data[SERVICE_ENERGY_TODAY].utcnow() + timedelta(hours=1)
+        ),
+    ),
+    EnergyZeroSensorEntityDescription(
+        key="average_price",
+        name="Average electricity market price - Today",
+        device_class=SensorDeviceClass.MONETARY,
+        native_unit_of_measurement=f"{CURRENCY_EURO}/{ENERGY_KILO_WATT_HOUR}",
+        value_fn=lambda data: data[SERVICE_ENERGY_TODAY].average_price,
+    ),
+    EnergyZeroSensorEntityDescription(
+        key="max_price",
+        name="Highest Price - Today",
+        device_class=SensorDeviceClass.MONETARY,
+        native_unit_of_measurement=f"{CURRENCY_EURO}/{ENERGY_KILO_WATT_HOUR}",
+        value_fn=lambda data: data[SERVICE_ENERGY_TODAY].extreme_prices[1],
+    ),
+    EnergyZeroSensorEntityDescription(
+        key="min_price",
+        name="Lowest Price - Today",
+        device_class=SensorDeviceClass.MONETARY,
+        native_unit_of_measurement=f"{CURRENCY_EURO}/{ENERGY_KILO_WATT_HOUR}",
+        value_fn=lambda data: data[SERVICE_ENERGY_TODAY].extreme_prices[0],
+    ),
+    EnergyZeroSensorEntityDescription(
+        key="highest_price_time",
+        name="Time of highest price - Today",
+        device_class=SensorDeviceClass.TIMESTAMP,
+        value_fn=lambda data: data[SERVICE_ENERGY_TODAY].highest_price_time,
+    ),
+    EnergyZeroSensorEntityDescription(
+        key="lowest_price_time",
+        name="Time of lowest price - Today",
+        device_class=SensorDeviceClass.TIMESTAMP,
+        value_fn=lambda data: data[SERVICE_ENERGY_TODAY].lowest_price_time,
+    ),
+    EnergyZeroSensorEntityDescription(
+        key="percentage_of_max",
+        name="Current percentage of highest electricity price - Today",
+        native_unit_of_measurement=PERCENTAGE,
+        icon="mdi:percent",
+        value_fn=lambda data: data[SERVICE_ENERGY_TODAY].pct_of_max_price,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up EnergyZero Sensors based on a config entry."""
+    coordinator: EnergyZeroDataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
+    entities: list[EnergyZeroSensorEntity] = []
+    entities.extend(
+        EnergyZeroSensorEntity(
+            coordinator=coordinator,
+            description=description,
+            name="Energy Market Price",
+            service="today_energy",
+        )
+        for description in SENSORS_ENERGY
+    )
+    if entry.data[CONF_GAS]:
+        entities.extend(
+            EnergyZeroSensorEntity(
+                coordinator=coordinator,
+                description=description,
+                name="Gas Market Price",
+                service="today_gas",
+            )
+            for description in SENSORS_GAS
+        )
+    async_add_entities(entities)
+
+
+class EnergyZeroSensorEntity(
```

Comment:
This seems to use the new entity style naming, but it doesn't set it as such?

Reply:
What do you mean exactly?

Reply:
See: <https://developers.home-assistant.io/blog/2022/07/10/entity_naming>